parser/src/tests/errors.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::ParseError::InvalidInt;
+use crate::{build_parse_test, error::Error, tests::ParseTest};
+
+// SCAN ERRORS
+// ================================================================================================
+
+#[test]
+fn error_invalid_token() {
+    let source = "
+    transition_constraints:
+        enf clk'' = clk + 1";
+    // Only one "'" is allowed at the end of identifier to indicate next row of the column.
+    // Should fail if more than one "'" is present in an identifier
+    let error = Error::ScanError(45..46);
+    build_parse_test!(source).expect_error(error);
+}
+
+#[test]
+fn error_identifier_with_invalid_characters() {
+    let source = "
+    transition_constraints:
+        enf clk@' = clk@ + 1";
```

Comment:
I would only use the `@` once. Otherwise once we start handling multiple errors this will need to be changed. I'd rather have this have a single error and then when we want to test multiple errors later we can add new tests for that.

If you want, you could instead add one new test with a source string that contains 2 errors to show that it fails at the first `ScanError` and only returns one error. Then when we add handling of multiple errors we'll expect that test to fail & we'll replace/update it

Reply:
Makes sense. Done.